Use formals() instead of args() in make_default_method_def() to address issue #25

const-ae opened the pull request:
`args()` does not set the enclosing environment of the function correctly
(i.e., it uses the GlobalEnv) which leads to problems because `callGeneric()` and `match.fun()`  cannot find the function.

This should fix #25.

const-ae committed:
Use formals() instead of args() in make_default_method_def()

args() does not set the enclosing environment of the function correctly
(i.e., it uses the GlobalEnv) which leads to problems because
callGeneric() then cannot find the function.

const-ae commented:
If nobody objects, I will merge this in the next days :)

PeteHaitch commented:
I don't fully understand what is happening here and defer to @hpages's expertise.

hpages commented:
Good catch @const-ae and your PR seems to solve issue #25. However, the environment of the default method still doesn't look right:
```
> environment(selectMethod(MatrixGenerics::colSums2, "ANY"))
<environment: 0x557123b06e88>
```
Should probably be `namespace:MatrixGenerics`, like for the non-default methods defined in the package:
```
> environment(selectMethod(MatrixGenerics::colSums2, "matrix_OR_array_OR_table_OR_numeric"))
<environment: namespace:MatrixGenerics>
```
So maybe it's just a matter for adding the following line in the `make_default_method_def()` function?
```
environment(def) <- getNamespace("MatrixGenerics")
```
Thanks!

const-ae committed:
Explicitly set environment of fallback function

Following the advise of @hpages (https://github.com/Bioconductor/MatrixGenerics/pull/26#issuecomment-870940312)

const-ae committed:
Bump version and update NEWS

const-ae commented:
Great. @hpages, thanks for the feedback. You are right, setting `environment(def) <- getNamespace("MatrixGenerics")` set the correct environment. I implemented the changes, updated the NEWS, and bumped the version.

PeteHaitch commented:
Thank you very much for handling this, @const-ae.

PeteHaitch commented:
I've pushed this to BioC's git server.
Does it need to be backported to the current release branch?

const-ae commented:
Thanks. No I don't think backporting is necessary. I assume this mainly affects package development and that should anyway be based on master :)

hpages commented:
Thanks guys. @const-ae Note that if you're explicitly setting the environment of the function then replacing `args()` with `formals()` was not needed.
